python/run.py

```diff
@@ -1,49 +1,133 @@
+
+credential = InteractiveBrowserCredential(tenant_id = tenantId)
+iotc = IotCentralClient(credential, subscriptionId)
 
-client = IotCentralClient(creds[0], subId)
 
-result = client.apps.check_name_availability(appName)
-print(result)
+# Choose app name.
 
-app = App(location="unitedstates", sku=AppSkuInfo(name="ST2"))
+appName = f"pysdk-{random.randint(100000, 999999)}-{random.randint(100000, 999999)}"
+
+
+# Define the app configuration.
+
+app = App(location = region, sku = AppSkuInfo(name = skuName))
 app.subdomain = appName
 app.display_name = appName
+app.identity = { "type": "SystemAssigned" }
 
-client.apps.create_or_update(resourceGroup, appName, app)
 
-getResult = client.apps.get(resourceGroup, appName)
-print(getResult)
+# Create the app.
 
-updateApp = AppPatch()
-updateApp.display_name = appName + "-new-name"
+print(f"\nCreating {appName}. Check browser window for login.")
+poller = iotc.apps.begin_create_or_update(resourceGroup, appName, app)
+result = poller.result()
 
-client.apps.update(resourceGroup, appName, updateApp)
+if result.state != "created":
+    raise Exception("Expected 'created' state.")
 
-appsInGroup = client.apps.list_by_resource_group(resourceGroup)
-appsInGroup.next()
-for item in appsInGroup.current_page:
-    print(item)
+print("OK")
+
+
+# Make sure it's idempotent.
+
+print(f"\nUpdating {appName}.")
+poller = iotc.apps.begin_create_or_update(resourceGroup, appName, app)
+result = poller.result()
+
+if result.state != "created":
+    raise Exception("Expected 'created' state.")
+
+print("OK")
 
-operations = client.operations.list()
-operations.next()
-for item in operations.current_page:
-    print(item)
 
-appTemplates = client.apps.list_templates()
-for item in appTemplates:
+# List all the apps in the resource group.
+
+print(f"\nListing IoT Central apps in '{resourceGroup}'")
+appsInGroup = iotc.apps.list_by_resource_group(resourceGroup)
+
+for item in appsInGroup:
     print(item)
 
-# deleteResult = client.apps.delete(resourceGroup, appName)
-# print(deleteResult)
 
-print("done")
+# Update the app tags.
+
+print(f"\nUpdating {appName} tags.")
+tag = "mytag"
+value = "myvalue"
+app.tags = { tag: value }
+poller = iotc.apps.begin_create_or_update(resourceGroup, appName, app)
+result = poller.result()
+
+if result.tags[tag] != value:
+    raise Exception("Expected updated tag.")
+
+print("OK")
+
+
+# Delete the app.
+
+print(f"\nDeleting {appName}")
+poller = iotc.apps.begin_delete(resourceGroup, appName)
+result = poller.result()
+
+if result:
+    print(result)
+    raise Exception("Expected 'None'.")
+
+print("OK")
+
+
+# Verify that we can't create in geography as we could before.
+
+print("\nMake sure we can't use geography.")
+app.location = "unitedstates"
+appName = "us-" + appName
+app.subdomain = app.display_name = appName
+
+try:
+    poller = iotc.apps.begin_create_or_update(resourceGroup, appName, app)
+    result = poller.result()
+    print("It worked but it shouldn't have!")
+
```

Comment:
add raise Exception here?

Reply:
This is inside a try who's catch which is the success condition and prints "OK", so raising here would print "OK".